README.md

```diff
@@ -1,16 +1,185 @@
+
+> "Take a bunch of raw data and output a re-organized version of it, or do a
+> calculation from all of it."
+
+In fact, the complex world of programming was often housed in a department
+called "data processing."
+
+Imagine you're back in that early era of "data processing." Let's now suppose
+someone asks us to:
+
+> Get the element in the `1` index for all the nested data.
+
+Thanks to the human-brain-friendly nested structure, we can look at the code
+and pretty quickly come up with `[3, 4]`. Since we know what the result should
+be thanks to "eyeballing it," coding this solution can be verified quickly:
+
+```ruby
+array_of_arrays = [ # containing Array (of Array)
+  [1, 3, 5], # "Nested" Array 1
+  [2, 4, 6]  # "Nested" Array 2
+]
+
+def find_firsts(src)
+  result = []
+  0.upto(src.count - 1) do |inner_count|
+    result << src[inner_count][1]
+  end
+  result
+end
+
+find_firsts(array_of_arrays) #=> [3, 4]
+```
+
+OK, we admit, determining the second odd or even number greater than zero isn't
+_that_ exciting. But what if each of these inner `Array`s represented three
+estimates, sorted least-to-most on remodeling two different rooms of your
+house? Suddenly this _exact same_ algorithm is much more exciting:
+
+```ruby
+array_of_bid_arrays = [ # containing Array (of Array)
+  [1000, 3000, 5000], # "Nested" Array 1
+  [2000, 4000, 6000]  # "Nested" Array 2
+]
+
+def what_we_can_get_away_with(src)
+  result = 0
+  0.upto(src.count - 1) do |inner_count|
+    result += src[inner_count][0]
+  end
+  result
+end
+
+def if_money_were_no_object(src)
+  result = 0
+  0.upto(src.count - 1) do |inner_count|
+    result += src[inner_count][2]
+  end
+  result
+end
+
+low_high_totals = [
+  what_we_can_get_away_with(array_of_bid_arrays),
+  if_money_were_no_object(array_of_bid_arrays)
+]
+
+p low_high_totals #=> [3000, 11000]
+p low_high_totals[1] - low_high_totals[1] #=> 8000
+```
+
+Wow! A difference of $8,000! The ability to take a nested structure and
+transform is to produce insight is how many programmers justify their salaries!
+
+## Nested Data Are Provided By Others for Us to Use
+
+As you learn more about software development, you'll want to get data from
+other sources to produce insights. Typically companies make this information
+available through an "API." For the time being, think of an API as a URL which
+your code can contact that returns....you guessed it...a nested data structure.
+
+Obviously, the data at Amazon is not curated by Jeff Bezos for the benefit of
+_my_ program. The latitude/longitude data of all the coffee shops near my
+iPhone's current location are not curated for the benefit of my program by Tim
+Cook and Apple. The information about my GitHub repositories is not curated by
+Nat Friedman for the benefit of _my_ program. The returned NDS _might_ have the
+data I need. But oftentimes, we need to transform this provided data in order
+to get the insights we need.
```

Comment:
this is all good, but my only thought is to maybe take out the people and only keep the companies